src/aiida/orm/implementation/storage_backend.py

```diff
@@ -304,6 +304,20 @@ def maintain(self, full: bool = False, dry_run: bool = False, **kwargs) -> None:
         :param dry_run: flag to only print the actions that would be taken without actually executing them.
         """
 
+    def backup(
```

Comment:
I realized that we don't check that the `dest` corresponds to the same storage if it already exists. Now you can specify the same destination folder using any number of different profiles and the code won't complain. Shouldn't we restrict a destination folder to a single profile?

I would implement that logic here and then define the `_backup` method that it calls. The subclasses then implement that method with the actual backup logic. This way we are guaranteed that all storage plugins will have the basic profile validation.